insights/tests/client/connection/test_checkin.py

```diff
@@ -191,3 +191,14 @@ def test_response_failure(get_proxies, post, get_canonical_facts, rm_conf, statu
 
     with raises(Exception):
         connection.checkin()
+
+
+def test_unregistered_host():
+    """
+    If the host is not registered, the checkin fails.
+    """
+    config = InsightsConfig(legacy_upload=False, base_url="www.example.com")
+    connection = InsightsConnection(config)
+
+    result = connection.checkin()
+    assert result is False
```

Comment:
I think there should be one more test verifying that the unregistered state of the host actually causes the checkin function to immediately return. This can be achieved by checking that `get_canonical_facts` and `post` aren’t called. Maybe split into two test cases. 🤷🏻‍♂️

```suggestion
    assert result is False


@patch('insights.client.connection.InsightsUploadConf.get_rm_conf', return_value={})
@patch("insights.client.connection.InsightsConnection._init_session", Mock())
@patch("insights.client.connection.post")
@patch("insights.client.connection.get_canonical_facts")
def test_unregistered_host_no_checkin(get_canonical_facts, post, rm_conf)
    config = InsightsConfig(legacy_upload=False, base_url="www.example.com")
    connection = InsightsConnection(config)

    connection.checkin()
    get_canonical_facts.assert_not_called()
    post.assert_not_called()
```

Reply:
I think one test for this is enough. However, I added there two asserts to ensure `get_canonical_facts` and `post` functions are not called :+1:

Reply:
Whatever. 🤷🏻‍♂️